### PR TITLE
feat(j-s): Enable file rename on case file screen

### PR DESCRIPTION
**Pull request by oddsson**
# Enable file rename on case file screen

[Asana](https://app.asana.com/0/1199153462262248/1203046355378034/f)

## What

Enable file rename on case file screen

## Screenshots / Gifs

https://user-images.githubusercontent.com/3789875/197753083-1a3cdb0f-f4e9-4d42-9eed-e89726169116.mov

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by gudjong — APPROVED** [review]

File: apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx [hunk 1]
```diff
@@ -195,55 +237,107 @@ const CaseFile: React.FC<CaseFileProps> = (props) => {
...
-              <Icon icon="pencil" color="blue400" />
-            </button>
+          <Box width="full">
+            <AnimatePresence initial={false} exitBeforeEnter>
+              {editFileId === caseFile.id ? (
+                <motion.div
+                  initial={{ y: 10, opacity: 0 }}
+                  animate={{ y: 0, opacity: 1 }}
+                  exit={{ y: 10, opacity: 0 }}
+                  transition={{ duration: 0.5 }}
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx:269]: Consider making this 200ms. Short animations should be under 200ms. Here is a good ref: https://m3.material.io/styles/motion/easing-and-duration/tokens-specs

File: apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx [hunk 2]
```diff
@@ -195,55 +237,107 @@ const CaseFile: React.FC<CaseFileProps> = (props) => {
...
+                      </button>
+                    </Box>
+                  </Box>
+                </motion.div>
+              ) : (
+                <motion.div
+                  initial={{ y: -10, opacity: 0 }}
+                  animate={{ y: 0, opacity: 1 }}
+                  exit={{ y: -10, opacity: 0 }}
+                  transition={{ duration: 0.5 }}
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx:301]: Same here

File: apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx [hunk 3]
```diff
@@ -361,6 +457,39 @@ const IndictmentsCaseFilesAccordionItem: React.FC<Props> = (props) => {
...
+    setReorderableItems((prev) => {
+      const newReorderableItems = [...prev]
+      newReorderableItems[
+        fileInReorderableItems
+      ].userGeneratedFilename = newName
+
+      return newReorderableItems
+    })
+
+    updateFilesMutation({
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx:478]: Consider showing toaster if rename of file fails.

File: apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx [hunk 4]
```diff
@@ -163,11 +172,44 @@ const renderChapter = (chapter: number, name: string) => (
...
-  const { caseFile, index, onReorder, onOpen } = props
+  const { caseFile, index, onReorder, onOpen, onRename } = props
+  const { formatMessage } = useIntl()
   const y = useMotionValue(0)
   const boxShadow = useRaisedShadow(y)
   const controls = useDragControls()
-  const [isDragging, setIsDragging] = useState(false)
+  const [isDragging, setIsDragging] = useState<boolean>(false)
+  const [editFileId, setEditFileId] = useState<string>()
+  const [editedFilenames, setEditedFilenames] = useState<EditedFileName[]>([])
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx:182]: I don't quite get why we need to keep `editedFilenames` in a state if we are not rendering them anywhere. Can we discuss this?

**Review by Arnarkari93 — CHANGES_REQUESTED** [review]

**Review by Arnarkari93 — APPROVED** [review]

**Review by RunarVestmann — APPROVED** [review]
Looks good to me

**Pull request merged**